sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java

```diff
@@ -853,15 +851,19 @@ public TextAnalyticsServiceVersion getServiceVersion() {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<DocumentSentiment> analyzeSentiment(String document, String language) {
-        Objects.requireNonNull(document, "'document' cannot be null.");
-        return analyzeSentimentBatch(Collections.singletonList(document), language, null)
-            .map(sentimentResult -> {
-                if (sentimentResult.isError()) {
-                    throw logger.logExceptionAsError(Transforms.toTextAnalyticsException(sentimentResult.getError()));
-                }
-
-                return sentimentResult.getDocumentSentiment();
-            }).last();
+        try {
+            Objects.requireNonNull(document, "'document' cannot be null.");
```

Comment:
Wouldn't it be better to make a call to inputDocumentsValidation here instead of calling Objects.requireNonNull? You can either make your inputDpcumentsValidation function take in strings as well as iterable of strings with an overload, or pass in the document in an array of 1

Reply:
'InputDocumentValidation' is a helper method for a list of documents, `documents`. It checks 'documents' is not null and not empty. But here, the input is a single document, 'document', so it won't apply here.